<?xml version="1.0"?>
<launch>
  <!-- turtlebot -->
  <include file="$(find turtlebot_bringup)/launch/minimal.launch" />

  <node  pkg="urg_node" type="urg_node" name="urg_node" >
    <param name="angle_min" value="-2.6"/>
    <param name="angle_max" value="2.6"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="laser_in_base_link"
    args="0.03 0.0 0.0 0.0 0.0 0.0 /base_link /laser 80" />

  <node pkg="tf" type="static_transform_publisher" name="base_link_in_odom"
    args="0.0 0.0 0.0 0.0 0.0 0.0 /odom /base_link 80" />

  <node pkg="torob" type="random_goal" name="goal" output="screen">
      <param name="duration" value="10"/>
      <param name="min_distance" value="0.5"/>
      <param name="max_distance" value="3"/>
      <param name="max_angle" value="1.8"/>
  </node>

  <node pkg="torob" type="ring_move" name="move" output="screen">
      <param name="robot_radius" value="0.2"/>
      <param name="safe_distance" value="0.8"/>
      <param name="ring_size" value="24"/>
      <param name="goal_frame_id" value="odom"/>
      <param name="init_goal_x" value="0" />
      <param name="init_goal_y" value="0" />
      <param name="linear_speed" value="0.2" />
      <param name="angular_speed" value="1.2" />
  </node>

  <node pkg="torob" type="viewer" name="scan_vmap_view" output="screen">
      <param name="vmap_topic" value="/scan_vmap"/>
      <param name="scan_topic" value="/scan"/>
      <param name="ring_topic" value="/ring_state"/>
      <param name="scale" value="240.0" />
      <param name="ring_size" value="13"/>
  </node>

</launch>
